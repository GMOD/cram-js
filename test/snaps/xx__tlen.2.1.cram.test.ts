import { testFile } from '../testUtil'
testFile('xx#tlen.2.1.cram')
