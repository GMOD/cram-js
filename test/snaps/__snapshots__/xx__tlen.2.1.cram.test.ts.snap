// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`can dump the whole xx#tlen.2.1.cram without error 1`] = `
[
  {
    "fileId": "-",
    "magic": "CRAM",
    "majorVersion": 2,
    "minorVersion": 1,
  },
  {
    "containerHeader": {
      "_endPosition": 39,
      "_size": 13,
      "alignmentSpan": 0,
      "landmarks": [
        0,
      ],
      "length": 10593,
      "numBases": 0,
      "numBlocks": 1,
      "numLandmarks": 1,
      "numRecords": 0,
      "recordCounter": 0,
      "refSeqId": 0,
      "refSeqStart": 0,
    },
    "data": [
      {
        "_endPosition": 39,
        "_size": 13,
        "alignmentSpan": 0,
        "landmarks": [
          0,
        ],
        "length": 10593,
        "numBases": 0,
        "numBlocks": 1,
        "numLandmarks": 1,
        "numRecords": 0,
        "recordCounter": 0,
        "refSeqId": 0,
        "refSeqStart": 0,
      },
      {
        "samHeader": [
          {
            "data": [
              {
                "tag": "SN",
                "value": "xx",
              },
              {
                "tag": "LN",
                "value": "20",
              },
              {
                "tag": "M5",
                "value": "bbf4de6d8497a119dda6e074521643dc",
              },
              {
                "tag": "UR",
                "value": "/home/vadim/bin/htslib/test/xx.fa",
              },
            ],
            "tag": "SQ",
          },
          {
            "data": [
              {
                "tag": "SN",
                "value": "yy",
              },
              {
                "tag": "LN",
                "value": "20",
              },
              {
                "tag": "M5",
                "value": "bbf4de6d8497a119dda6e074521643dc",
              },
              {
                "tag": "UR",
                "value": "/home/vadim/bin/htslib/test/xx.fa",
              },
            ],
            "tag": "SQ",
          },
          {
            "data": [
              {
                "tag": "xx has been encoded using the SAM spec; leftmost to rightmost",
                "value": "",
              },
            ],
            "tag": "CO",
          },
          {
            "data": [
              {
                "tag": "yy has been encoded using bwa/picard methods; 5' to 3'",
                "value": "",
              },
            ],
            "tag": "CO",
          },
          {
            "data": [
              {
                "tag": "00000000011111111112",
                "value": "",
              },
            ],
            "tag": "CO",
          },
          {
            "data": [
              {
                "tag": "12345678901234567890",
                "value": "",
              },
            ],
            "tag": "CO",
          },
          {
            "data": [
              {
                "tag": "AAAAAAAAAATTTTTTTTTT",
                "value": "",
              },
            ],
            "tag": "CO",
          },
          {
            "data": [
              {
                "tag": "1>>>>          <<<<2   x1",
                "value": "",
              },
            ],
            "tag": "CO",
          },
          {
            "data": [
              {
                "tag": "",
                "value": "",
              },
            ],
            "tag": "CO",
          },
          {
            "data": [
              {
                "tag": "      1>>>>    ",
                "value": "",
              },
              {
                "tag": "       x2",
                "value": "",
              },
            ],
            "tag": "CO",
          },
          {
            "data": [
              {
                "tag": "         <<<<2",
                "value": "",
              },
            ],
            "tag": "CO",
          },
          {
            "data": [
              {
                "tag": "",
                "value": "",
              },
            ],
            "tag": "CO",
          },
          {
            "data": [
              {
                "tag": "         1>>>>",
                "value": "",
              },
              {
                "tag": "       x3 (7..15 vs 10..11)",
                "value": "",
              },
            ],
            "tag": "CO",
          },
          {
            "data": [
              {
                "tag": "      <<<<2",
                "value": "",
              },
            ],
            "tag": "CO",
          },
          {
            "data": [
              {
                "tag": "",
                "value": "",
              },
            ],
            "tag": "CO",
          },
          {
            "data": [
              {
                "tag": "<<<<2          1>>>>   x4 (1..20 vs 16..5)",
                "value": "",
              },
            ],
            "tag": "CO",
          },
        ],
      },
    ],
  },
  {
    "compressionScheme": {
      "APdelta": false,
      "dataSeriesEncoding": {
        "AP": {
          "codecId": 6,
          "parameters": {
            "length": 5,
            "offset": 0,
          },
          "parametersBytes": 2,
        },
        "BF": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              1,
              1,
            ],
            "numCodes": 2,
            "numLengths": 2,
            "symbols": [
              99,
              147,
            ],
          },
          "parametersBytes": 7,
        },
        "CF": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              1,
              1,
            ],
            "numCodes": 2,
            "numLengths": 2,
            "symbols": [
              1,
              5,
            ],
          },
          "parametersBytes": 6,
        },
        "FN": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "IN": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 13,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "MQ": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              1,
            ],
          },
          "parametersBytes": 4,
        },
        "NF": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "QS": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 12,
          },
          "parametersBytes": 1,
        },
        "RG": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              -1,
            ],
          },
          "parametersBytes": 8,
        },
        "RI": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "RL": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              5,
            ],
          },
          "parametersBytes": 4,
        },
        "RN": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 11,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "SC": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 14,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "TL": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
      },
      "readNamesIncluded": true,
      "referenceRequired": false,
      "substitutionMatrix": [
        [
          "C",
          "G",
          "T",
          "N",
        ],
        [
          "A",
          "G",
          "T",
          "N",
        ],
        [
          "A",
          "C",
          "T",
          "N",
        ],
        [
          "A",
          "C",
          "G",
          "N",
        ],
        [
          "A",
          "C",
          "G",
          "T",
        ],
      ],
      "tagEncoding": {},
      "tagIdsDictionary": [
        [],
      ],
    },
    "containerHeader": {
      "_endPosition": 10646,
      "_size": 14,
      "alignmentSpan": 20,
      "landmarks": [
        140,
      ],
      "length": 281,
      "numBases": 40,
      "numBlocks": 5,
      "numLandmarks": 1,
      "numRecords": 8,
      "recordCounter": 0,
      "refSeqId": 0,
      "refSeqStart": 1,
    },
    "data": [
      {
        "_endPosition": 10646,
        "_size": 14,
        "alignmentSpan": 20,
        "landmarks": [
          140,
        ],
        "length": 281,
        "numBases": 40,
        "numBlocks": 5,
        "numLandmarks": 1,
        "numRecords": 8,
        "recordCounter": 0,
        "refSeqId": 0,
        "refSeqStart": 1,
      },
      {
        "features": [
          {
            "alignmentStart": 1,
            "cramFlags": 5,
            "flags": 99,
            "lengthOnRef": 5,
            "mappingQuality": 1,
            "mate": {
              "alignmentStart": 16,
              "readName": "x1",
              "sequenceId": 0,
              "uniqueId": 10793,
            },
            "mateRecordNumber": undefined,
            "qualityScores": [
              9,
              9,
              9,
              9,
              9,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 5,
            "readName": "x1",
            "sequenceId": 0,
            "tags": {},
            "templateLength": 20,
            "templateSize": undefined,
            "uniqueId": 10792,
          },
          {
            "alignmentStart": 16,
            "cramFlags": 1,
            "flags": 147,
            "lengthOnRef": 5,
            "mappingQuality": 1,
            "mate": {
              "alignmentStart": 1,
              "readName": "x1",
              "sequenceId": 0,
              "uniqueId": 10792,
            },
            "qualityScores": [
              9,
              9,
              9,
              9,
              9,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 5,
            "readName": "x1",
            "sequenceId": 0,
            "tags": {},
            "templateLength": 20,
            "templateSize": undefined,
            "uniqueId": 10793,
          },
          {
            "alignmentStart": 7,
            "cramFlags": 5,
            "flags": 99,
            "lengthOnRef": 5,
            "mappingQuality": 1,
            "mate": {
              "alignmentStart": 10,
              "readName": "x2",
              "sequenceId": 0,
              "uniqueId": 10795,
            },
            "mateRecordNumber": undefined,
            "qualityScores": [
              9,
              9,
              9,
              9,
              9,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 5,
            "readName": "x2",
            "sequenceId": 0,
            "tags": {},
            "templateLength": 8,
            "templateSize": undefined,
            "uniqueId": 10794,
          },
          {
            "alignmentStart": 10,
            "cramFlags": 1,
            "flags": 147,
            "lengthOnRef": 5,
            "mappingQuality": 1,
            "mate": {
              "alignmentStart": 7,
              "readName": "x2",
              "sequenceId": 0,
              "uniqueId": 10794,
            },
            "qualityScores": [
              9,
              9,
              9,
              9,
              9,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 5,
            "readName": "x2",
            "sequenceId": 0,
            "tags": {},
            "templateLength": 8,
            "templateSize": undefined,
            "uniqueId": 10795,
          },
          {
            "alignmentStart": 7,
            "cramFlags": 5,
            "flags": 147,
            "lengthOnRef": 5,
            "mappingQuality": 1,
            "mate": {
              "alignmentStart": 10,
              "readName": "x3",
              "sequenceId": 0,
              "uniqueId": 10797,
            },
            "mateRecordNumber": undefined,
            "qualityScores": [
              9,
              9,
              9,
              9,
              9,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 5,
            "readName": "x3",
            "sequenceId": 0,
            "tags": {},
            "templateLength": 8,
            "templateSize": undefined,
            "uniqueId": 10796,
          },
          {
            "alignmentStart": 10,
            "cramFlags": 1,
            "flags": 99,
            "lengthOnRef": 5,
            "mappingQuality": 1,
            "mate": {
              "alignmentStart": 7,
              "readName": "x3",
              "sequenceId": 0,
              "uniqueId": 10796,
            },
            "qualityScores": [
              9,
              9,
              9,
              9,
              9,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 5,
            "readName": "x3",
            "sequenceId": 0,
            "tags": {},
            "templateLength": 8,
            "templateSize": undefined,
            "uniqueId": 10797,
          },
          {
            "alignmentStart": 1,
            "cramFlags": 5,
            "flags": 147,
            "lengthOnRef": 5,
            "mappingQuality": 1,
            "mate": {
              "alignmentStart": 16,
              "readName": "x4",
              "sequenceId": 0,
              "uniqueId": 10799,
            },
            "mateRecordNumber": undefined,
            "qualityScores": [
              9,
              9,
              9,
              9,
              9,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 5,
            "readName": "x4",
            "sequenceId": 0,
            "tags": {},
            "templateLength": 20,
            "templateSize": undefined,
            "uniqueId": 10798,
          },
          {
            "alignmentStart": 16,
            "cramFlags": 1,
            "flags": 99,
            "lengthOnRef": 5,
            "mappingQuality": 1,
            "mate": {
              "alignmentStart": 1,
              "readName": "x4",
              "sequenceId": 0,
              "uniqueId": 10798,
            },
            "qualityScores": [
              9,
              9,
              9,
              9,
              9,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 5,
            "readName": "x4",
            "sequenceId": 0,
            "tags": {},
            "templateLength": 20,
            "templateSize": undefined,
            "uniqueId": 10799,
          },
        ],
        "header": {
          "_endPosition": 10821,
          "_size": 30,
          "compressedSize": 30,
          "compressionMethod": "raw",
          "content": Uint8Array [
            0,
            1,
            20,
            8,
            0,
            3,
            2,
            11,
            12,
            255,
            255,
            255,
            255,
            15,
            187,
            244,
            222,
            109,
            132,
            151,
            161,
            25,
            221,
            166,
            224,
            116,
            82,
            22,
            67,
            220,
          ],
          "contentId": 0,
          "contentPosition": 10791,
          "contentType": "MAPPED_SLICE_HEADER",
          "parsedContent": {
            "_endPosition": 10676,
            "_size": 30,
            "contentIds": [
              11,
              12,
            ],
            "md5": [
              187,
              244,
              222,
              109,
              132,
              151,
              161,
              25,
              221,
              166,
              224,
              116,
              82,
              22,
              67,
              220,
            ],
            "numBlocks": 3,
            "numContentIds": 2,
            "numRecords": 8,
            "recordCounter": 0,
            "refBaseBlockId": -1,
            "refSeqId": 0,
            "refSeqSpan": 20,
            "refSeqStart": 1,
          },
          "uncompressedSize": 30,
        },
      },
    ],
  },
  {
    "compressionScheme": {
      "APdelta": false,
      "dataSeriesEncoding": {
        "AP": {
          "codecId": 6,
          "parameters": {
            "length": 5,
            "offset": 0,
          },
          "parametersBytes": 2,
        },
        "BF": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              1,
              1,
            ],
            "numCodes": 2,
            "numLengths": 2,
            "symbols": [
              99,
              147,
            ],
          },
          "parametersBytes": 7,
        },
        "CF": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              1,
              2,
              2,
            ],
            "numCodes": 3,
            "numLengths": 3,
            "symbols": [
              3,
              1,
              5,
            ],
          },
          "parametersBytes": 8,
        },
        "FN": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "IN": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 13,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "MF": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              1,
              1,
            ],
            "numCodes": 2,
            "numLengths": 2,
            "symbols": [
              0,
              1,
            ],
          },
          "parametersBytes": 6,
        },
        "MQ": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              1,
            ],
          },
          "parametersBytes": 4,
        },
        "NF": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "NP": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              2,
              2,
              2,
              2,
            ],
            "numCodes": 4,
            "numLengths": 4,
            "symbols": [
              1,
              7,
              10,
              16,
            ],
          },
          "parametersBytes": 10,
        },
        "NS": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              1,
            ],
          },
          "parametersBytes": 4,
        },
        "QS": {
          "codecId": 1,
          "parameters": {
            "blockContentId": 12,
          },
          "parametersBytes": 1,
        },
        "RG": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              -1,
            ],
          },
          "parametersBytes": 8,
        },
        "RI": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              1,
            ],
          },
          "parametersBytes": 4,
        },
        "RL": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              5,
            ],
          },
          "parametersBytes": 4,
        },
        "RN": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 11,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "SC": {
          "codecId": 5,
          "parameters": {
            "blockContentId": 14,
            "stopByte": 0,
          },
          "parametersBytes": 2,
        },
        "TL": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              0,
            ],
            "numCodes": 1,
            "numLengths": 1,
            "symbols": [
              0,
            ],
          },
          "parametersBytes": 4,
        },
        "TS": {
          "codecId": 3,
          "parameters": {
            "bitLengths": [
              2,
              2,
              2,
              2,
            ],
            "numCodes": 4,
            "numLengths": 4,
            "symbols": [
              -10,
              -2,
              2,
              10,
            ],
          },
          "parametersBytes": 18,
        },
      },
      "readNamesIncluded": true,
      "referenceRequired": false,
      "substitutionMatrix": [
        [
          "C",
          "G",
          "T",
          "N",
        ],
        [
          "A",
          "G",
          "T",
          "N",
        ],
        [
          "A",
          "C",
          "T",
          "N",
        ],
        [
          "A",
          "C",
          "G",
          "N",
        ],
        [
          "A",
          "C",
          "G",
          "T",
        ],
      ],
      "tagEncoding": {},
      "tagIdsDictionary": [
        [],
      ],
    },
    "containerHeader": {
      "_endPosition": 10941,
      "_size": 14,
      "alignmentSpan": 20,
      "landmarks": [
        197,
      ],
      "length": 341,
      "numBases": 40,
      "numBlocks": 5,
      "numLandmarks": 1,
      "numRecords": 8,
      "recordCounter": 8,
      "refSeqId": 1,
      "refSeqStart": 1,
    },
    "data": [
      {
        "_endPosition": 10941,
        "_size": 14,
        "alignmentSpan": 20,
        "landmarks": [
          197,
        ],
        "length": 341,
        "numBases": 40,
        "numBlocks": 5,
        "numLandmarks": 1,
        "numRecords": 8,
        "recordCounter": 8,
        "refSeqId": 1,
        "refSeqStart": 1,
      },
      {
        "features": [
          {
            "alignmentStart": 1,
            "cramFlags": 5,
            "flags": 99,
            "lengthOnRef": 5,
            "mappingQuality": 1,
            "mate": {
              "alignmentStart": 16,
              "readName": "y1",
              "sequenceId": 1,
              "uniqueId": 11153,
            },
            "mateRecordNumber": undefined,
            "qualityScores": [
              9,
              9,
              9,
              9,
              9,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 5,
            "readName": "y1",
            "sequenceId": 1,
            "tags": {},
            "templateLength": 20,
            "templateSize": undefined,
            "uniqueId": 11152,
          },
          {
            "alignmentStart": 16,
            "cramFlags": 1,
            "flags": 147,
            "lengthOnRef": 5,
            "mappingQuality": 1,
            "mate": {
              "alignmentStart": 1,
              "readName": "y1",
              "sequenceId": 1,
              "uniqueId": 11152,
            },
            "qualityScores": [
              9,
              9,
              9,
              9,
              9,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 5,
            "readName": "y1",
            "sequenceId": 1,
            "tags": {},
            "templateLength": 20,
            "templateSize": undefined,
            "uniqueId": 11153,
          },
          {
            "alignmentStart": 7,
            "cramFlags": 5,
            "flags": 99,
            "lengthOnRef": 5,
            "mappingQuality": 1,
            "mate": {
              "alignmentStart": 10,
              "readName": "y2",
              "sequenceId": 1,
              "uniqueId": 11155,
            },
            "mateRecordNumber": undefined,
            "qualityScores": [
              9,
              9,
              9,
              9,
              9,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 5,
            "readName": "y2",
            "sequenceId": 1,
            "tags": {},
            "templateLength": 8,
            "templateSize": undefined,
            "uniqueId": 11154,
          },
          {
            "alignmentStart": 10,
            "cramFlags": 1,
            "flags": 147,
            "lengthOnRef": 5,
            "mappingQuality": 1,
            "mate": {
              "alignmentStart": 7,
              "readName": "y2",
              "sequenceId": 1,
              "uniqueId": 11154,
            },
            "qualityScores": [
              9,
              9,
              9,
              9,
              9,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 5,
            "readName": "y2",
            "sequenceId": 1,
            "tags": {},
            "templateLength": 8,
            "templateSize": undefined,
            "uniqueId": 11155,
          },
          {
            "alignmentStart": 7,
            "cramFlags": 3,
            "flags": 147,
            "lengthOnRef": 5,
            "mappingQuality": 1,
            "mate": {
              "alignmentStart": 10,
              "flags": 0,
              "readName": undefined,
              "sequenceId": 1,
            },
            "qualityScores": [
              9,
              9,
              9,
              9,
              9,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 5,
            "readName": "y3",
            "sequenceId": 1,
            "tags": {},
            "templateSize": -2,
            "uniqueId": 11156,
          },
          {
            "alignmentStart": 10,
            "cramFlags": 3,
            "flags": 99,
            "lengthOnRef": 5,
            "mappingQuality": 1,
            "mate": {
              "alignmentStart": 7,
              "flags": 1,
              "readName": undefined,
              "sequenceId": 1,
            },
            "qualityScores": [
              9,
              9,
              9,
              9,
              9,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 5,
            "readName": "y3",
            "sequenceId": 1,
            "tags": {},
            "templateSize": 2,
            "uniqueId": 11157,
          },
          {
            "alignmentStart": 1,
            "cramFlags": 3,
            "flags": 147,
            "lengthOnRef": 5,
            "mappingQuality": 1,
            "mate": {
              "alignmentStart": 16,
              "flags": 0,
              "readName": undefined,
              "sequenceId": 1,
            },
            "qualityScores": [
              9,
              9,
              9,
              9,
              9,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 5,
            "readName": "y4",
            "sequenceId": 1,
            "tags": {},
            "templateSize": 10,
            "uniqueId": 11158,
          },
          {
            "alignmentStart": 16,
            "cramFlags": 3,
            "flags": 99,
            "lengthOnRef": 5,
            "mappingQuality": 1,
            "mate": {
              "alignmentStart": 1,
              "flags": 1,
              "readName": undefined,
              "sequenceId": 1,
            },
            "qualityScores": [
              9,
              9,
              9,
              9,
              9,
            ],
            "readBases": undefined,
            "readGroupId": -1,
            "readLength": 5,
            "readName": "y4",
            "sequenceId": 1,
            "tags": {},
            "templateSize": -10,
            "uniqueId": 11159,
          },
        ],
        "header": {
          "_endPosition": 11173,
          "_size": 30,
          "compressedSize": 30,
          "compressionMethod": "raw",
          "content": Uint8Array [
            1,
            1,
            20,
            8,
            8,
            3,
            2,
            11,
            12,
            255,
            255,
            255,
            255,
            15,
            187,
            244,
            222,
            109,
            132,
            151,
            161,
            25,
            221,
            166,
            224,
            116,
            82,
            22,
            67,
            220,
          ],
          "contentId": 0,
          "contentPosition": 11143,
          "contentType": "MAPPED_SLICE_HEADER",
          "parsedContent": {
            "_endPosition": 10971,
            "_size": 30,
            "contentIds": [
              11,
              12,
            ],
            "md5": [
              187,
              244,
              222,
              109,
              132,
              151,
              161,
              25,
              221,
              166,
              224,
              116,
              82,
              22,
              67,
              220,
            ],
            "numBlocks": 3,
            "numContentIds": 2,
            "numRecords": 8,
            "recordCounter": 8,
            "refBaseBlockId": -1,
            "refSeqId": 1,
            "refSeqSpan": 20,
            "refSeqStart": 1,
          },
          "uncompressedSize": 30,
        },
      },
    ],
  },
]
`;
